Packs/DigitalGuardian/Integrations/DigitalGuardianARCEventCollector/DigitalGuardianARCEventCollector.py

```diff
@@ -22,239 +24,246 @@ class Client(BaseClient):
+            resp_type='text',
         )
-        return response
 
 
-def test_module(client: Client, params: Dict[str, Any]) -> str:
+def test_module(client: Client, params: dict[str, Any]) -> str:
     """
-    Tests API connectivity and authentication'
-    When 'ok' is returned it indicates the integration works like it is supposed to and connection to the service is
-    successful.
-    Raises exceptions if something goes wrong.
+    Tests API connectivity and authentication.
     Args:
         client (Client): Digital Guardian client to use.
-        params (Dict): Integration parameters.
+        params (dict): Integration parameters.
     Returns:
-        str: 'ok' if test passed, anything else will raise an exception and will fail the test.
+        str: 'ok' connection to the service is successful.
+    Raises:
+        DemistoException | HTTPError: If request failed.
     """
     try:
-        limit = arg_to_number(params.get('number_of_events')) or 1000
-        fetch_events(
-            client=client,
-            last_run={},
-            limit=limit,
-        )
+        client.get_watchlists()
+        return 'ok'
 
     except Exception as e:
         if 'Forbidden' in str(e):
-            raise DemistoException('Authorization Error: make sure API Key is correctly set')
-        else:
-            raise e
-    return "ok"
+            raise DemistoException('Authorization Error: Make sure client credentials are correctly set') from e
+        raise
 
 
-def get_raw_events(client: Client, time_of_last_event: str) -> list:
-    """
-       helper function that is used in get-events and fetch-events to get the actual events and sort them
-       Args:
-           client (Client): DG client
-           time_of_last_event: time of last ingested event
-       Returns:
-           list: list of events
-    """
-    outcome = []
-    event_list = []
-    if not time_of_last_event:
-        time_of_last_event_datetime = datetime.now() - timedelta(hours=1)
-        time_of_last_event_str = datetime_to_string(time_of_last_event_datetime)
-    else:
-        temp_time: datetime = arg_to_datetime(arg=time_of_last_event, required=True)  # type: ignore[assignment]
-        time_of_last_event_str = temp_time.isoformat(sep=' ', timespec='milliseconds')
-    current_time = datetime_to_string(datetime.now())
-    events = client.get_events(time_of_last_event_str, current_time)
-    events = {} if events is None else events
-    for field_names in events.get("fields"):
-        outcome.append(field_names.get('name'))
-    for event in events.get("data"):
-        result = dict(zip(outcome, event))
-        event_list.append(result)
-    event_list.sort(key=lambda item: (item.get("inc_mtime"), item.get("dg_guid")))
-    return event_list
-
-
-def get_events_command(client: Client, args: dict) -> Tuple[list, CommandResults]:
+def fetch_events(client: Client, limit: int | None = None) -> tuple[EventsGenerator, dict]:
```

Comment:
gotta love generators 🌟 

here, let's return a list/tuple of events, to simplify the process and keep things a bit more standard (devs debugging this code won't expect a generator, but the actual events)